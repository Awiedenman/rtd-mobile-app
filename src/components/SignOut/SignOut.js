import React from 'react';

import { auth } from '../../firebase';

const SignOutButton = () => {
  return (
    <button
      type="button"
      onClick={auth.doSignOut}
    >
      Sign Out
    </button>
  );
};

export default SignOutButton;