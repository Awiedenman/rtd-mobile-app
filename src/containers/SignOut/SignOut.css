#sign-out-container {
  position: absolute;
  bottom: 100px;
}