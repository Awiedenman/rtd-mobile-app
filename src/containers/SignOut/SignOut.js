import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signOutUser } from '../../actions';

import { auth } from '../../firebase';

import './SignOut.css';

const SignOutButton = (props) => {
  const signOut = () => {
    auth.doSignOut();
    props.signOutUser();
  };

  return (
    <button
      className='sign-out-button'
      type="button"
      onClick={signOut}
    >
      Sign Out
    </button>
  );
};

export const mapDispatchToProps = dispatch => ({
  signOutUser: () => dispatch(signOutUser())
});

export default connect(null, mapDispatchToProps)(SignOutButton);

SignOutButton.propTypes = {
  signOutUser: PropTypes.func
};