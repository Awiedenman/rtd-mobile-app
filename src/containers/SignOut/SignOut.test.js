import React from 'react';
import { shallow } from 'enzyme';
import { SignOutButton } from '../SignOut/SignOut';

describe('SingOutButton', () => {
  it('should match Snapshot', () => {
    const mockSignOutUser = jest.fn();
    const wrapper = shallow(
      <SignOutButton 
        signOutUser={ mockSignOutUser }
      />);
      
    expect(wrapper).toMatchSnapshot();
  });
});

