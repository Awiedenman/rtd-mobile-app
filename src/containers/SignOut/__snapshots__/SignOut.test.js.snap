// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingOutButton should match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    signOutUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
>
          To sign out...
</h3>,
        <button
          className="button"
          onClick={[Function]}
          type="button"
>
          Sign Out
</button>,
      ],
      "className": "form",
      "id": "sign-out-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To sign out...",
          "className": "title",
        },
        "ref": null,
        "rendered": "To sign out...",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Out",
          "className": "button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
>
            To sign out...
</h3>,
          <button
            className="button"
            onClick={[Function]}
            type="button"
>
            Sign Out
</button>,
        ],
        "className": "form",
        "id": "sign-out-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To sign out...",
            "className": "title",
          },
          "ref": null,
          "rendered": "To sign out...",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Out",
            "className": "button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
